synopsys-task/src/main.ts

```diff
@@ -72,19 +83,33 @@ export async function run() {
   console.log("Synopsys Task workflow execution completed");
 }
 
-export function logBridgeExitCodes(message: string): string {
-  const exitCode = message.trim().slice(-1);
+export function logExitCodes(message: string, exitCode: string): string {
   return constants.EXIT_CODE_MAP.has(exitCode)
     ? "Exit Code: " + exitCode + " " + constants.EXIT_CODE_MAP.get(exitCode)
-    : message;
+    : "Undefined error from extension: "
+        .concat(message)
+        .concat(constants.SPACE)
+        .concat(ErrorCode.UNDEFINED_ERROR_FROM_EXTENSION.toString());
 }
 
 run().catch((error) => {
   if (error.message != undefined) {
     taskLib.error(error.message);
+    const isReturnStatusEnabled = parseToBoolean(inputs.RETURN_STATUS);
+    const status = error.message.trim().split(" ").pop() || "";
+
+    // The statement set the exit code in the 'status' variable which can be used in the YAML file
+    if (isReturnStatusEnabled) {
+      console.log(
+        `##vso[task.setvariable variable=status;isoutput=true]${status}`
```

Comment:
can we cover this in test cases if possible,like getting ADO exit codes and returning based on return status, or it is not required?